screen-orientation/lock-basic.html

```diff
@@ -27,17 +26,27 @@
     'landscape-secondary',
     'portrait-primary',
   ];
-  const promises = orientations.map(type => {
-    const returnType = screen.orientation.lock(type);
+  for (const orientation of orientations) {
+    const returnType = screen.orientation.lock(orientation);
     assert_true(returnType instanceof Promise);
-  });
-  return Promise.all(promises).finally(() => {
-    screen.orientation.unlock();
-  });
+    await returnType;
+  }
+  screen.orientation.unlock();
 }, "Test that screen.orientation.lock returns a pending promise.");
 
-test(() => {
-  const previousOrientation = screen.orientation;
-  assert_equals(screen.orientation, previousOrientation);
+promise_test(async t => {
+  const preType = screen.orientation.type;
+  let newType = "";
+
+  if (preType.indexOf("portrait") != -1) {
+    newType = "landscape-primary";
+  } else {
+    newType = "portrait-primary";
+  }
+
+  const p = screen.orientation.lock(newType);
+  assert_equals(screen.orientation.type, preType);
```

Comment:
Nit: good to explain what is being tested here.
```JS
assert_equals(screen.orientation.type, preType, "Must not change orientation until next spin of event loop");
```

Reply:
Updated.